stingray/io.py

```diff
@@ -653,3 +660,66 @@ def get_btis(gtis, start_time=None, stop_time=None):
 def gti_len(gti):
     """Return the total good time from a list of GTIs."""
     return np.sum([g[1] - g[0] for g in gti])
+
+def _save_pickle_object(object, filename):
+    pickle.dump(object, open(filename, "wb" ))
+
+def _retrieve_pickle_object(filename):
+    return pickle.load(open(filename, "rb" ) )
+
+def _save_hdf5_object(object, filename):
+    pass
+
+def _retrieve_hdf5_object(object, filename):
+    pass
+
+def _save_ascii_object(object, filename):
+    pass
+
+def _retrieve_ascii_object(object, filename):
+    pass
+
+def save_object(object, filename, format):
+    """
+    Pickle a class instance.
+
+    Parameters
+    ----------
+    object: a class instance
```

Comment:
While implementing tests, I have come across a major problem of pickle. That is, nesting classes makes pickle fail. The reason, as I have discovered is, pickle relies on the path of object inside the application to reconstruct it later. This is, however, fixed in Python 3. So, currently the tests that I have added run on python 3 but fail on python 2.

So, I think we would be better off making some other format standard. json is an option - faster, lighter etc. Suggestions?